superset/connectors/base/models.py

```diff
@@ -166,7 +166,9 @@ def select_star(self):
     def data(self):
         """Data representation of the datasource sent to the frontend"""
         order_by_choices = []
-        for s in sorted(self.column_names):
+        # self.column_names return sorted column_names
+        for s in self.column_names:
+            s = str(s or '')
```

Comment:
I think your investigation surfaced that the are columns without a name which seems bad. I can look into why there are columns without names and produce a migration if needed.

I'm a little perplexed what `order_by_choices` means for empty strings and how this is used.